sdk/typescript/runtime/introspector.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"path/filepath"
+	"typescript-sdk/internal/dagger"
+	"typescript-sdk/tsdistconsts"
+)
+
+const (
+	introspectorBinPath = "/bin/ts-introspector"
+	typescriptLibPath   = "/src/node_modules/typescript"
+)
+
+type Introspector struct {
+	//+private
+	Ctr *dagger.Container
+}
+
+func NewIntrospector(sdkSourceDir *dagger.Directory) *Introspector {
+	ctr := dag.
+		Container().
+		From(tsdistconsts.DefaultBunImageRef).
+		WithMountedFile(introspectorBinPath, sdkSourceDir.File(introspectorBinPath)).
+		WithMountedDirectory(typescriptLibPath, sdkSourceDir.Directory("typescript-library")).
+		WithWorkdir(ModSourceDirPath)
+
+	return &Introspector{
+		Ctr: ctr,
+	}
+}
+
+func (i *Introspector) AsEntrypoint(
+	outputFilePath string,
+
+	moduleName string,
+
+	sourceCode *dagger.Directory,
+
+	clientBindings *dagger.File,
+) *dagger.Container {
+	return i.Ctr.
+		WithDirectory(ModSourceDirPath, sourceCode, dagger.ContainerWithDirectoryOpts{
```

Comment:
nit: if this can be `WithMountedDirectory` that would be a bit more efficient by avoiding copying

Reply:
Done [f0b76ec](https://github.com/dagger/dagger/pull/11267/commits/f0b76ec99c361a230a3e04d47aef1a9d74ac04f9)